### DISCUSSION TITLE
Loading assemblies onto a Cloud Instance

### DISCUSSION THREAD

**Discussion by georryan** [ravendb/ravendb#19860] [Q&A] [answered]
I've been playing with RavenDB on my laptop and have made a handful of indexes.  I've been working on a Mac.  I've just opened up a free cloud instance on AWS and have successfully made a new database and am able to send and receive data to and from it.  I tried to run my program that would update the indexes and it failed.  The error said the DLL I was pointing to didn't exist.  I then tried to create one of the indexes manually in the studio and ran into the same problem. I thought maybe it was the path from a Mac that was causing the problem, and tried the same thing from my windows box.  I'm still getting an error (although now it is an error about the absolute path).  What am I missing here?  I can successfully create indexes on a locally hosted database, without issue.  Do I need to upload my DLLs somewhere in the cloud and set the path differently somehow?  I feel like I am just missing a small detail in order to pull this all together.

**Comment by ayende** [comment, marked as answer]
What is the actual error that you are getting?

In general, you _can_ use assemblies in your cloud instances, but you need to use the Additional Assemblies feature and use a `NuGet` repository for those.

Note that you don't have to use the public one, but can use your own if needed

![image](https://github.com/user-attachments/assets/cb4788d9-f93d-4fe7-9f26-503c2838348f)

**Reply by georryan** [reply]
The error is related to the path I submitted for the assembly.  Is it not possible to use the path option for an assembly on a cloud server instance?  I had assumed doing that would upload the assembly to the server, but maybe not?  

If NuGet is the only option, then it isn't possible to run an additional assembly on a cloud instance without a professional license? I'm using the community license on the free tier database at the moment. I'm trying to test out a mobile app concept.

Thanks for responding!

**Reply by ayende** [reply]
You cannot upload a file to the server, no. NuGet is the only supported way to do that.

The reason behind that is that you _have_ to have the same version across all nodes, And our internal systems can do _things_ to the OS such that files you upload will not be there. But a structure approach to package management will work.

**Reply by georryan** [reply]
That makes sense. It's not clear from the screen that those restrictions are in place, though.  Thanks for explaining.  I'll reach out to sales and see what a professional license will cost so that I can create my own nuget package and use that solution.  I'll need to create an API layer soon and having a cloud instance will be useful.

**Reply by ayende** [reply]
You can also see here: https://cloud.ravendb.net/pricing

**Reply by georryan** [reply]
Thanks!  I saw that earlier when setting up my server originally, but looking closer at it I think I understand the restrictions a little better now.  If I'm reading this right, the free AWS tier is limited to a community license. So, it can't be upgraded to a professional or cloud license.  To do that I'd have to go with a paid cloud instance on one of the hosting options.  Am I on the right track there?

**Reply by ayende** [reply]
Yes, that is correct.
